pkg/controller/datavolume-controller.go

```diff
@@ -1379,10 +1379,11 @@ func (r *DatavolumeReconciler) expand(log logr.Logger,
 	}
 
 	expansionRequired := actualSize.Cmp(requestedSize) < 0
+	updateRequestSizeRequired := actualSize.Cmp(requestedSize) <= 0 && requestedSize.Cmp(currentSize) > 0
```

Comment:
I guess you haven't used expansionRequired because you want to have also the equals case?
I think the second comparison will be better readable as  currentSize.Cmp(requestedSize) <= 0, similar to the first one.

Reply:
right the expansion required is used later in the code, adding the = to it changes that logic. Ill change as you suggest

Reply:
actually changing to currentSize.Cmp(requestedSize) <= 0 is wrong since you dont need to update if they are equal.. so ill change it to < if its more clear that the requested size is the second parameter in the comparison